src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -387,89 +380,161 @@ bool SQLStorage::loadMetadata(Uptane::MetaPack* metadata) {
-    }
-    metadata->director_root = Uptane::Root("director", fixed_json);
-  } else {
+  const std::string* data = &req_response_table[0]["director_root"];
+  if (data->empty()) {
     return false;
+  } else if (metadata) {
+    metadata->director_root = *data;
   }
 
-  json = Utils::parseJSON(req_response_table[0]["image_root"]);
-  if (json != Json::nullValue) {
-    Json::Value fixed_json = json;
-    if (!json.isMember("signed")) {
-      fixed_json["signed"] = json;
-    }
-    metadata->image_root = Uptane::Root("image", fixed_json);
-  } else {
+  data = &req_response_table[0]["image_root"];
+  if (data->empty()) {
     return false;
+  } else if (metadata) {
+    metadata->image_root = *data;
   }
 
-  json = Utils::parseJSON(req_response_table[0]["director_targets"]);
-  if (json != Json::nullValue) {
-    Json::Value fixed_json = json;
-    if (!json.isMember("signed")) {
-      fixed_json["signed"] = json;
-    }
-    metadata->director_targets = Uptane::Targets(fixed_json);
+  data = &req_response_table[0]["director_targets"];
+  if (data->empty()) {
+    return false;
+  } else if (metadata) {
+    metadata->director_targets = *data;
   }
 
-  json = Utils::parseJSON(req_response_table[0]["image_targets"]);
-  if (json != Json::nullValue) {
-    Json::Value fixed_json = json;
-    if (!json.isMember("signed")) {
-      fixed_json["signed"] = json;
-    }
-    metadata->image_targets = Uptane::Targets(fixed_json);
+  data = &req_response_table[0]["image_targets"];
+  if (data->empty()) {
+    return false;
+  } else if (metadata) {
+    metadata->image_targets = *data;
   }
 
-  json = Utils::parseJSON(req_response_table[0]["image_timestamp"]);
-  if (json != Json::nullValue) {
-    Json::Value fixed_json = json;
-    if (!json.isMember("signed")) {
-      fixed_json["signed"] = json;
-    }
-    metadata->image_timestamp = Uptane::TimestampMeta(fixed_json);
-  }
+  data = &req_response_table[0]["image_timestamp"];
+  if (data->empty())
+    return false;
+  else if (metadata)
+    metadata->image_timestamp = *data;
 
-  json = Utils::parseJSON(req_response_table[0]["image_snapshot"]);
-  if (json != Json::nullValue) {
-    Json::Value fixed_json = json;
-    if (!json.isMember("signed")) {
-      fixed_json["signed"] = json;
-    }
-    metadata->image_snapshot = Uptane::Snapshot(fixed_json);
+  data = &req_response_table[0]["image_snapshot"];
+  if (data->empty()) {
+    return false;
+  } else if (metadata) {
+    metadata->image_snapshot = *data;
   }
 
   return true;
 }
 
-void SQLStorage::clearMetadata() {
+void SQLStorage::clearMetadataCommon(const std::string& tablename) {
   SQLite3Guard db(config_.sqldb_path.c_str());
 
   if (db.get_rc() != SQLITE_OK) {
     LOG_ERROR << "Can't open database: " << db.errmsg();
     return;
   }
 
-  if (db.exec("DELETE FROM meta;", nullptr, nullptr) != SQLITE_OK) {
+  if (db.exec("DELETE FROM " + tablename + ";", nullptr, nullptr) != SQLITE_OK) {
```

Comment:
Should use `db.prepareStatement`.

Also, providing an exec method which takes a `std::string` (https://github.com/advancedtelematic/aktualizr/pull/777/files#diff-8323b86c37db8f6551a01f8aec6344d6R119) encourages this kind of string concatenation business, so I'd also recommend not adding it.

Reply:
Hum sorry, my bad, this might be impossible. I'll take another look.

Reply:
Yes, I believe it's impossible unfortunately. It is marginally OK in my opinion, because the function only gets called with hardcoded constants.

Reply:
yes, not the end of the world